src/NServiceBus.Core/Persistence/InMemory/SagaPersister/InMemorySagaPersister.cs

```diff
@@ -87,26 +87,28 @@ public void Update(IContainSagaData saga)
         void ValidateUniqueProperties(IContainSagaData saga)
         {
             var sagaMetaData = sagaModel.FindByEntityName(saga.GetType().FullName);
-
-            if (!sagaMetaData.CorrelationProperties.Any()) return;
-
-            var sagasFromSameType = from s in data
-                                    where
-                                        (s.Value.SagaEntity.GetType() == saga.GetType() && (s.Key != saga.Id))
-                                    select s.Value;
-
-            foreach (var storedSaga in sagasFromSameType)
+            var existingSagas = (from s in data
+                where s.Value.SagaEntity.GetType() == saga.GetType() && (s.Key != saga.Id)
+                select s.Value)
+                .ToList();
```

Comment:
Not sure if it makes sense, but because in the `foreach` u can have an early exit (https://github.com/Particular/NServiceBus/pull/2653/files#diff-17e8d72500928dcb599bc890835658eeR114), should we not cast to list ? And let it be lazy evaluated?